src/views/BridgeTransactionsHistory.vue

```diff
@@ -177,29 +177,29 @@ export default class BridgeTransactionsHistory extends Mixins(TranslationMixin,
     return true
   }
 
-  async showHistory (idOrHash: string): Promise<void> {
-    const currentTransaction = this.filteredHistory.filter(item => (item.id === idOrHash) || (this.isOutgoingType(item.type) ? item.hash === idOrHash : item.ethereumHash === idOrHash))[0]
-    if (currentTransaction) {
+  async showHistory (id: string): Promise<void> {
+    const tx = api.bridge.getHistory(id)
+    if (tx) {
       await this.setTransactionConfirm(true)
-      await this.setSoraToEthereum(this.isOutgoingType(currentTransaction.type))
-      await this.setAssetAddress(this.getAssetBySymbol(currentTransaction?.symbol)?.address)
-      await this.setAmount(currentTransaction.amount)
-      await this.setSoraTransactionHash(currentTransaction.hash)
-      await this.setSoraTransactionDate(currentTransaction[this.isOutgoingType(currentTransaction.type) ? 'startTime' : 'endTime'])
-      await this.setEthereumTransactionHash(currentTransaction.ethereumHash)
-      await this.setEthereumTransactionDate(currentTransaction[!this.isOutgoingType(currentTransaction.type) ? 'startTime' : 'endTime'])
-      if (currentTransaction.status === BridgeTxStatus.Failed) {
+      await this.setSoraToEthereum(this.isOutgoingType(tx.type))
+      await this.setAssetAddress(this.getAssetBySymbol(tx.symbol || '')?.address)
```

Comment:
May we replace it with `.......byAddress` function?

Reply:
The same thing as was mentioned above.